pkgs/applications/editors/rstudio/default.nix

```diff
@@ -1,115 +1,183 @@
-{ lib, mkDerivation, fetchurl, fetchpatch, fetchFromGitHub, makeDesktopItem, cmake, boost, zlib
-, openssl, R, qtbase, qtxmlpatterns, qtsensors, qtwebengine, qtwebchannel
-, libuuid, hunspellDicts, unzip, ant, jdk, gnumake, makeWrapper, pandoc
+{ lib
+, mkDerivation
+, fetchurl
+, fetchpatch
+, fetchFromGitHub
+, makeDesktopItem
+, cmake
+, boost
+, zlib
+, openssl
+, R
+, qtbase
+, qtxmlpatterns
+, qtsensors
+, qtwebengine
+, qtwebchannel
+, libuuid
+, hunspellDicts
+, unzip
+, ant
+, jdk
+, gnumake
+, makeWrapper
+, pandoc
 , llvmPackages
+, libyamlcpp
+, soci
+, postgresql
+, nodejs
+, mkYarnPackage
+, qmake
 }:
 
-with lib;
 let
-  verMajor = "1";
-  verMinor = "2";
-  verPatch = "5042";
-  version = "${verMajor}.${verMinor}.${verPatch}";
-  ginVer = "2.1.2";
-  gwtVer = "2.8.1";
+  mathJaxSrc = fetchurl {
+    url = "https://s3.amazonaws.com/rstudio-buildtools/mathjax-27.zip";
+    sha256 = "sha256-xWy6psTOA8H8uusrXqPDEtL7diajYCVHcMvLiPsgQXY=";
+  };
+
+  rsconnectSrc = fetchFromGitHub {
+    owner = "rstudio";
+    repo = "rsconnect";
+    rev = "f5854bb71464f6e3017da9855f058fe3d5b32efd";
+    sha256 = "sha256-ULyWdSgGPSAwMt0t4QPuzeUE6Bo6IJh+5BMgW1bFN+Y=";
+  };
 in
 mkDerivation rec {
   pname = "RStudio";
-  inherit version;
-
-  nativeBuildInputs = [ cmake unzip ant jdk makeWrapper pandoc ];
-
-  buildInputs = [ boost zlib openssl R qtbase qtxmlpatterns qtsensors
-                  qtwebengine qtwebchannel libuuid ];
+  RSTUDIO_VERSION_MAJOR = "1";
+  RSTUDIO_VERSION_MINOR = "4";
+  RSTUDIO_VERSION_PATCH = "1717";
+  version = "${RSTUDIO_VERSION_MAJOR}.${RSTUDIO_VERSION_MINOR}.${RSTUDIO_VERSION_PATCH}";
```

Comment:
Why not use lib.versions.major/minor/majorMinor/etc?

Reply:
i didn't know there was a versions.patch 
i switched them to 
```
  version = "1.4.1717";
  RSTUDIO_VERSION_MAJOR = lib.versions.major version;
  RSTUDIO_VERSION_MINOR = lib.versions.minor version; 
  RSTUDIO_VERSION_PATCH = lib.versions.patch version;
```

Reply:
The RStudio build process uses those envvars: E.g. https://github.com/rstudio/rstudio/blob/de82b4fb8b6138221d735508c8bfb55fbb3e1b03/CMakeGlobals.txt#L29